src/subdomains/supporting/bank/bank-tx/bank-tx.service.ts

```diff
@@ -66,6 +73,19 @@ export class BankTxService {
     if (olkyTransactions.length > 0) await this.settingService.set(settingKeyOlky, newModificationTime);
   }
 
+  async assignTransactions() {
+    const unassignedBankTx = await this.bankTxRepo.find({ where: { type: IsNull() } });
+
+    for (const tx of unassignedBankTx) {
+      const match = /^[0-9A-Z]{4}-[0-9A-Z]{4}-[0-9A-Z]{4}$/.test(tx.remittanceInfo);
+
+      if (match) {
+        const buy = await this.buyService.getBuyRepo().findOneBy({ bankUsage: tx.remittanceInfo });
+        if (buy) await this.update(tx.id, { type: BankTxType.BUY_CRYPTO, buyId: buy.id });
```

Comment:
Maybe add a `getByBankUsage` method to buy service and remove the `getBuyRepo`?

Reply:
You need to find by the regex match. Means you need to remove the start/end conditions on the regex and adapt the check to
```ts
const match = /[0-9A-Z]{4}-[0-9A-Z]{4}-[0-9A-Z]{4}/.exec(tx.remittanceInfo);
if (match) {
   const buy = await this.buyService.getBuyRepo().findOneBy({ bankUsage: match[0] });
   ...
}
```